lib/mock-module/mock-module.ts

```diff
@@ -21,15 +19,30 @@ const mockProvider = (provider: any): Provider | undefined => {
   const provide = typeof provider === 'object' && provider.provide ? provider.provide : provider;
   const multi = typeof provider === 'object' && provider.multi;
 
-  // RouterModule injects own Application Initializer, which doesn't allow mocks due to own logic.
-  // We have to avoid any injection of Application Initializer to mock everything properly.
+  // RouterModule injects own providers, which doesn't allow mocks due to own logic.
+  // We have to avoid any injection of those providers to mock everything properly.
   if (
     typeof provide === 'object' && provide.ngMetadataName === 'InjectionToken'
     && provide.toString() === 'InjectionToken Application Initializer'
   ) {
     return undefined;
   }
 
+  // BrowserAnimationsModule injects own providers, which doesn't allow mocks due to own logic.
+  // We have to avoid any injection of those providers to mock everything properly.
+  if (typeof provide === 'function' && provide.name === 'RendererFactory2') {
+    return undefined;
+  }
+
+  // BrowserModule injects own providers, which doesn't allow mocks due to own logic.
+  // We have to avoid any injection of those providers to mock everything properly.
+  if (typeof provide === 'function' && provide.name === 'ApplicationInitStatus') {
+    return undefined;
+  }
+  if (typeof provide === 'function' && provide.name === 'DomRendererFactory2') {
```

Comment:
Do you mind squashing the conditionals into one?
```suggestion
  if (typeof provide === 'function' && ['ApplicationInitStatus', 'DomRendererFactory2', ' RendererFactory2'].includes(provide.name) {
```

Reply:
Updated